.github/workflows/pr-title.yml

```
@@ -0,0 +1,101 @@
+name: PR Title Checker
+
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+
+jobs:
+  validate-title:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check PR Title Format
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const title = context.payload.pull_request.title.trim();
+            
+            // Define valid patterns
+            const validPatterns = [
+              // Pattern 1: (type) O3-XXXX: Description
+              /^\((docs|test|chore|fix|feat|refactor)\)\s+(O3-\d+):\s+.+$/,
+              // Pattern 2: (type) Description (no ticket)
+              /^\((docs|test|chore|fix|feat|refactor)\)\s+[^ ].+$/,
+              // Pattern 3: (BREAKING) O3-XXXX: Description
+              /^\(BREAKING\)\s+(O3-\d+):\s+.+$/,
+              // Pattern 4: BREAKING: Description (no ticket)
+              /^BREAKING:\s+[^ ].+$/
+            ];
+
+            // Check if title matches any valid pattern
+            const isValid = validPatterns.some(pattern => pattern.test(title));
+            
+            if (!isValid) {
+              let errorMessage = `🚫 Invalid PR title format. Strict requirements:\n\n`;
+              
+              // Check for common mistakes
+              if (/^BREAKING\s/i.test(title)) {
+                errorMessage += `• For BREAKING changes with ticket numbers, use "(BREAKING) O3-1234: Description"\n`;
+                errorMessage += `• For BREAKING changes without ticket numbers, use "BREAKING: Description"\n`;
+              } else if (/^\([^)]+\)\s+O3-\d+[^:]/.test(title)) {
+                errorMessage += `• Ticket reference must be followed by a colon (O3-1234: Description)\n`;
+              } else if (/^[^(](docs|test|chore|fix|feat|refactor)/i.test(title)) {
```

No need to repeat the whole thing. We're just checking that the start of the line is a parenthesis:

```suggestion
              } else if (/^[^(]/(/i.test(title)) {
```

---

But, of course, we actually want to check that there is at least one pair of parentheses.